### PR TITLE
samples: drivers: spi_flash rename definitions

### PR DESCRIPTION
**Pull request by FRASTM**
This commits prefixes all the definitions
with SPI_  to be used by the sample application _samples/drivers/spi_flash_ 

In that spi_flash sample, the #define are identified with SPI_FLASH_xxx and this will avoid conflicts with other `FLASH_SECTOR_SIZE` used by any driver or hal elsewhere.
Typically stm32h7b3xx defines FLASH_SECTOR_SIZE for itself.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
rebase, fix merge conflct

**Review by erwango — DISMISSED** [review]

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: samples/drivers/spi_flash/src/main.c [hunk 1]
```diff
@@ -71,8 +71,8 @@ void main(void)
 	printf("\nTest 1: Flash erase\n");
 
 	/* full flash erase if FLASH_TEST_REGION_OFFSET = 0  FLASH_SECTOR_SIZE = flash size */
-	rc = flash_erase(flash_dev, FLASH_TEST_REGION_OFFSET,
-			 FLASH_SECTOR_SIZE);
+	rc = flash_erase(flash_dev, SPI_FLASH_TEST_REGION_OFFSET,
```
[the remark this answers is not in the document]
Comment (FRASTM) [reply samples/drivers/spi_flash/src/main.c:76]: done

**Comment by FRASTM** [comment]
rebase
update comment

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Pull request merged**